kabuki.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+# Main kabuki.py
+# imports - use the rpi-rgb-led-matrix instructions to install dependencies
+import queue
+import time
+from multiprocessing import JoinableQueue
+from multiprocessing import Process
+
+from PIL import Image
+from PIL import ImageDraw
+from rgbmatrix import graphics
+
+from face import Face
+
+# constants used throughout project
+WIDTH = 32
+HEIGHT = 32
+SPEED = 1.0 / 20.0
+IDLE_TIME = 50
+EYES = 0
+MOUTH = 16
+
+
+class Kabuki:
+    def __init__(self, matrix):
+        self.matrix = matrix
+        self.face = Face()
+        self.eye_queue = JoinableQueue()
+        self.mouth_queue = JoinableQueue()
+        self.render_queue = JoinableQueue()
+        self.loop_proc = Process(target=self.loop)
+        self.compute_proc = Process(target=self.compute_frame)
+        self.render_proc = Process(target=self.render_loop)
+        self.reset_face()
+
+    def start_render(self):
+        self.loop_proc.start()
+        # self.loop_proc.join()
+        self.compute_proc.start()
+        self.render_proc.start()
+        # self.compute_proc.join()
+        # self.render_proc.join()
+
+    def play_seq(self, sequence, board):
+        for frame in sequence:
+            # self.matrix.SetImage(frame, 0, board)
+            if board == EYES:
+                self.eye_queue.put(frame)
+            else:
+                self.mouth_queue.put(frame)
+            # time.sleep(SPEED)
+
+    def play_hold(self, hold, board):
+        for frame in range(0, IDLE_TIME):
+            # self.matrix.SetImage(hold, 0, board)
+            if board == EYES:
+                self.eye_queue.put(frame)
+            else:
+                self.mouth_queue.put(frame)
+            # time.sleep(SPEED)
+
+    def compute_frame(self):
+        while 1:
+            # do work to combine eye and mouth and add to render queue
+            self.eye_queue.join()
+            self.mouth_queue.join()

```

Comment:
Not needed. You are waiting needlessly for the queues to join. Just having get() and put() are enough.